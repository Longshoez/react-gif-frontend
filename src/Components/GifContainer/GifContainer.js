import React from 'react';
import 'animate.css'
import PropTypes from 'prop-types';


const GifContainer = ({title, url, id}) => {
  return(
    <div key={id} className='gifComponent animate__animated animate__fadeIn'>
        {/**<h2 className='gifTitle'>{(title.length > 25) ? (title.substring(0, 25) + "...") : (title)}</h2>*/}
        <h2 className='gifTitle'>{title}</h2>
        <img className='gifImage' src={url} alt={title} />
    </div>
  );
};

GifContainer.propTypes = {
    title: PropTypes.string.isRequired,  
    url: PropTypes.string.isRequired,
}

export default GifContainer