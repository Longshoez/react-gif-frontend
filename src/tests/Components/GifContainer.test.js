import React from 'react';
import {shallow} from 'enzyme';
import GifContainer from '../../Components/GifContainer/GifContainer';

describe('Pruebas en el componente GifContainer', () => {
    
    const title ="a title"
    const url ="https://localhost/somesome.jpg"
    
    let wrapper = shallow(<GifContainer title={title} url={url}/>)

    // beforeEach(() => {
    //     wraper = shallow(<GifContainer />)
    // })

    test('Renderizar componente', () => {
        //Arrarnge
        //wraper declaerd in upper scope

        //Act
        //Assert
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe tener un h2 en el titulo', () => {
        //Arrange        
        
        //Act        
        const value = wrapper.find('h2')

        //Assert
        expect(value.text().trim()).toBe(title)
    });

    test('Debe tener un url y un alt en el elemento Img', () => {
        //Arrange        
        
        //Act        
        const image = wrapper.find('img')
        //console.log(image.props()); //props (by enzyme) lets you evaluate the properties of certain html element
        //Assert
        expect(image.prop('src')).toBe(url)
        expect(image.prop('alt')).toBe(title)
    });

    test('Debe tener la clase de animacion', () => {
        //Arrange        
        
        //Act        
        const div = wrapper.find('div')
        //console.log(div.prop('className')); //props (by enzyme) lets you evaluate the properties of certain html element

        //Assert
        expect(div.prop('className')).toContain('animate__animated')
        expect(div.prop('className')).toContain('animate__fadeIn')
        //course version
        const className = div.prop('className')
        expect(className.includes('animate__fadeIn')).toBe(true)
    });
})

