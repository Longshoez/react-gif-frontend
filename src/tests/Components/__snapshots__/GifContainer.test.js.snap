// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifContainer Renderizar componente 1`] = `
<div
  className="gifComponent animate__animated animate__fadeIn"
>
  <h2
    className="gifTitle"
  >
    a title
  </h2>
  <img
    alt="a title"
    className="gifImage"
    src="https://localhost/somesome.jpg"
  />
</div>
`;
